submissions/crealime/memory-pair-game/js/main.js

```diff
@@ -0,0 +1,255 @@
+            <li class="start-page__li"><a href="#" data-pair="10" class="start-page__link">10</a></li>
+            <li class="start-page__li"><a href="#" data-pair="12" class="start-page__link">12</a></li>
+            <li class="start-page__li"><a href="#" data-pair="15" class="start-page__link">15</a></li>
+            <li class="start-page__li"><a href="#" data-pair="18" class="start-page__link">18</a></li>
+            <li class="start-page__li"><a href="#" data-pair="21" class="start-page__link">21</a></li>
+            <li class="start-page__li"><a href="#" data-pair="28" class="start-page__link">28</a></li>
+          </ul>
+        </nav>
+      </div>
+      <label class="start-page__label">
+        <input type="checkbox" class="start-page__checkbox">
+        <span class="start-page__preview">Preview</span>
+      </label>
+    </div>
+  `
+}
+
+function getCardTemplate([url, id]) {
+  return `
+    <div class="game__card flip id-${id}" data-id="${id}">
+      <div class="game__front">
+        <img src="${url}" alt="" class="game__img">
+      </div>
+      <div class="game__back"></div>
+    </div>
+  `
+}
+
+function getResultPageTemplate(time, pairs) {
+  return `
+    <div class="result-page">
+      <div class="result-page__message">Your result is<br>${time}<br>for ${pairs} pairs</div>
+       ${resultMessage}
+      <a href="#" class="result-page__link">Play again</a>
+    </div>
+  `
+}
+
+function getResultMessageTemplate(isBest) {
+  return !isBest
+    ? `<div class="result-page__message_success">It’s your own record</div>`
+    : `<div class="result-page__message_warning">Your record is ${isBest}</div>`
+}
+
+function setTimeInLocaleStorage() {
+  const bestTime = localStorage.getItem(`pairs-${numOfPairs}`) || null;
+  if (!bestTime || bestTime && gameTime.getDiffTime().replace(/\D/g, '') < bestTime.replace(/\D/g, '')) {
+    localStorage.setItem(`pairs-${numOfPairs}`, gameTime.getDiffTime())
+    resultMessage = getResultMessageTemplate(false)
+  }
+  else {
+    resultMessage = getResultMessageTemplate(bestTime)
+  }
+}
+
+function fadeOutFadeIn(foo, num, preview) {
+  const animation = main.animate([
+    {opacity: 1},
+    {opacity: 0}
+  ], {
+    duration: TIMER.time,
+    easing: 'ease-out'
+  })
+
+  animation.addEventListener('finish', function() {
+    foo(num, preview)
+    main.animate([
+      {opacity: 0},
+      {opacity: 1}
+    ], {
+      duration: TIMER.time,
+      easing: 'ease-in'
+    })
+  })
+}
+
+function showPreviewCards() {
+  const cards = document.querySelectorAll('.game__card')
+
+  setTimeout(() => {
+    cards.forEach((card, i) => setTimeout(() => {
+      card.classList.toggle('flip')
+      card.classList.toggle('pointer')
+    }, i * TIMER.step))
+    cards.forEach((card, i) => setTimeout(() => {
+      card.classList.toggle('flip')
+      card.classList.toggle('pointer')
+    }, TIMER.time * numOfPairs + i * TIMER.step))
+  }, TIMER.time)
+}
+
+function flipCard(e) {
```

Comment:
Users can abuse the game by flipping the same card to get info about the position of all cards. Prevent clicking on an already opened card.

https://user-images.githubusercontent.com/16196199/193999874-9cb2ca4a-1046-4975-b660-748f5b31a434.mp4

Reply:
This was done on purpose, since the game is played on time, not on the number of clicks. Opening and closing the same card spending same time than opening two cards. But we get position of one card, not two. This cannot be abused.

Reply:
So, is this change necessary, or can I leave it the way I intended?